code/game/objects/effects/anomalies.dm

```diff
@@ -184,6 +184,11 @@
 /obj/effect/anomaly/grav/high/proc/setup_grav_field()
 	grav_field = new(src, 7, TRUE, rand(0, 3))
 
+/obj/effect/anomaly/grav/high/detonate()
+	for(var/obj/machinery/gravity_generator/main/the_generator in GLOB.machines)
+		if(is_station_level(the_generator.z))
```

Comment:
with GLOB.gravity_generators we can check for specific Z levels, so gravity anomalies on lower levels of maps wouldnt hit the grav gen on the higher ones. If it is something we'd want.

Reply:
This would make anomaly detonations do nothing on multi-Z stations and be confusing for players.